api/authenticator.py

```diff
@@ -665,38 +666,42 @@ def create_authentication_document(self):
             value = ConfigurationSetting.for_library(rel, library).value
             if not value:
                 continue
-            links[rel] = dict(href=value)
+            link = dict(rel=rel, href=value)
             if any(value.startswith(x) for x in ('http:', 'https:')):
                 # We assume that HTTP URLs lead to HTML, but we don't
                 # assume anything about other URL schemes.
-                links[rel]['type'] = "text/html"
-
+                link['type'] = "text/html"
+            links.append(link)
+                
         # Add a rel="help" link for every type of URL scheme that
         # leads to library-specific help.
         for type, uri in Configuration.help_uris(library):
-            link = dict(href=uri)
-            links.setdefault("help", []).append(link)
-            if type:
-                link['type'] = type
+            links.append(dict(rel="help", href=uri, type=type))
 
         # Add a link to the web page of the library itself.
         library_uri = ConfigurationSetting.for_library(
             Configuration.WEBSITE_URL, library).value
         if library_uri:
-            links['alternate'] = dict(type="text/html", href=library_uri)
+            links.append(
+                dict(rel="alternate", type="text/html", href=library_uri)
+            )
 
         # Add the library's logo, if it has one.
         logo = ConfigurationSetting.for_library(
             Configuration.LOGO, library).value
         if logo:
-            links["logo"] = dict(type="image/png", href=logo)
+            links.append(dict(rel="logo", type="image/png", href=logo))
                 
         library_name = self.library_name or unicode(_("Library"))
-        library_url = url_for("index", library_short_name=library.short_name, _external=True)
-        doc = OPDSAuthenticationDocument.fill_in(
-            base_opds_document, list(self.providers),
-            _db=self._db, name=library_name, id=library_url, links=links
+        auth_doc_url = url_for(
+            "authentication_document", library_short_name=library.short_name,
+            _external=True
```

Comment:
The `id` of an A4OPDS document is the URL to the document itself, not to the root of the OPDS server.